examples/with-okta-multi-tenant-pkce-flow/src/index.tsx

```diff
@@ -0,0 +1,119 @@
+
+import SuperTokens, { SuperTokensWrapper, getSuperTokensRoutesForReactRouterDom } from "supertokens-auth-react";
+import ThirdParty from "supertokens-auth-react/recipe/thirdparty";
+
+import Session from "supertokens-auth-react/recipe/session";
+
+import Index from "./routes/index"
+import Dashboard from "./routes/dashboard"
+import { PreAndPostAPIHookAction } from 'supertokens-web-js/recipe/thirdparty';
+
+SuperTokens.init({
+    appInfo: {
+        appName: "Demo",
+        // TODO Update as per your app's Website and API domains
+        apiDomain: "http://multitenant.com:8000",
+        websiteDomain: "http://multitenant.com:3000",
+    },
+    recipeList: [
+        ThirdParty.init({
+            signInAndUpFeature: {
+                providers: [
+                  { 
+                    // Adding a custom provider with id "okta" as we don't have support out of the box. Also adding a custom button component for the provider.
+                    // ref: https://supertokens.com/docs/thirdparty/common-customizations/sign-in-and-up/custom-providers
+                    id: "okta",
+                    name: "Okta",
+                    buttonComponent: <div style={{
+                        cursor: "pointer",
+                        border: "1",
+                        paddingTop: "5px",
+                        paddingBottom: "5px",
+                        borderRadius: "5px",
+                        borderStyle: "solid"
+                    }}>Login with Okta</div>
+                  }
+                ]
+            },
+            getRedirectionURL: async (context) => {
+              // This override ensure we redirect the logged in user to the dashboard page after login
+              if (context.action === "SUCCESS") {
+                  if (context.redirectToPath !== undefined) {
+                      // we are navigating back to where the user was before they authenticated
+                      return context.redirectToPath;
+                  }
+                  return "/dashboard";
+              }
+              return undefined;
+          },
+          preAPIHook: async (context) => {
+            // ref: https://supertokens.com/docs/thirdparty/advanced-customizations/frontend-hooks/pre-api
+            // As we don't have support for multi-tenancy out of the box, we need to pass additional 
+            // information to the APIs, and that can be done by overriding preAPIHook. Refer the comments
+            // in the code below to know more.
+
+            let url = context.url;
+            let requestInit = context.requestInit;
+            let action : PreAndPostAPIHookAction = context.action;
+
+            // Fetching the user entered TenantID from the local storage. This is set in the login route "/"
+            const tenant : String | null = window.localStorage.getItem('tenant');
+
+            // Adding the `state` and the `TenantID` as query parameters to the API calls.
```

Comment:
```suggestion
            // Adding the `state` and the `TenantID` as query parameters to the API calls so that the backend can consume them to get the correct Okta config for this tenant.
```

Reply:
done